common/tokensParser.go

```diff
@@ -33,6 +33,10 @@ func ExtractTokenIDAndNonceFromTokenStorageKey(tokenKey []byte) ([]byte, uint64)
 		return tokenKey, 0
 	}
 
+	if strings.Count(token, separatorChar) == 2 && string(token[len(token)-1]) != separatorChar {
+		token = token[indexOfFirstHyphen+1:]
```

Comment:
TokenID will be prefix+token - thus token should be token[:indexOfSecondHyphen]

Reply:
Yes, refactored code to work with existing tokens without prefix and the ones with prefix